examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImSubscribeCommand.java

```diff
@@ -0,0 +1,113 @@
+    @Override
+    public void onError(ChipAttributePath attributePath, ChipEventPath eventPath, Exception e) {
+      logger.log(Level.INFO, "subscribe receive onError");
+      setFailure("write failure");
+    }
+
+    @Override
+    public void onReport(NodeState nodeState) {
+      logger.log(Level.INFO, "onReport received ");
+    }
+  }
+
+  private class InternalGetConnectedDeviceCallback implements GetConnectedDeviceCallback {
+    @Override
+    public void onDeviceConnected(long devicePointer) {
+      setDevicePointer(devicePointer);
+      logger.log(Level.INFO, "onDeviceConnected");
+    }
+
+    @Override
+    public void onConnectionFailure(long nodeId, Exception error) {
+      logger.log(Level.INFO, "onConnectionFailure");
+    }
+  }
+
+  private class InternalSubscriptionEstablishedCallback implements SubscriptionEstablishedCallback {
+    @Override
+    public void onSubscriptionEstablished(long subscriptionId) {
+      logger.log(Level.INFO, "onSubscriptionEstablished with Id" + subscriptionId);
+      setSubscriptionId(subscriptionId);
+      setSuccess();
+    }
+  }
+
+  private class InternalResubscriptionAttemptCallback implements ResubscriptionAttemptCallback {
+    @Override
+    public void onResubscriptionAttempt(int terminationCause, int nextResubscribeIntervalMsec)
+    {
+      logger.log(Level.INFO, "ResubscriptionAttemptCallback");
+    }
+  }
+
+  public PairOnNetworkLongImSubscribeCommand(
+      ChipDeviceController controller, CredentialsIssuer credsIssue) {
+    super(
+        controller,
+        "onnetwork-long-im-subscribe",
+        PairingModeType.ON_NETWORK,
+        PairingNetworkType.NONE,
+        credsIssue,
+        DiscoveryFilterType.LONG_DISCRIMINATOR);
+  }
+
+  @Override
+  protected void runCommand() {
+    ChipAttributePath attributePath =
+    ChipAttributePath.newInstance(
+        /* endpointId= */ 0, CLUSTER_ID_BASIC, ATTR_ID_LOCAL_CONFIG_DISABLED);
+ArrayList<ChipAttributePath> attributePathList = new ArrayList<>();
+attributePathList.add(attributePath);
+    currentCommissioner()
+        .pairDeviceWithAddress(
+            getNodeId(),
+            getRemoteAddr().getHostAddress(),
+            MATTER_PORT,
+            getDiscriminator(),
+            getSetupPINCode(),
+            null);
+    currentCommissioner().setCompletionListener(this);
+    waitCompleteMs(getTimeoutMillis());
+    currentCommissioner()
+        .getConnectedDevicePointer(getNodeId(), new InternalGetConnectedDeviceCallback());
+    clear();
+
+    currentCommissioner()
+        .subscribeToPath(new InternalSubscriptionEstablishedCallback(), new InternalResubscriptionAttemptCallback(), new InternalReportCallback(), devicePointer, attributePathList, Collections.emptyList(), 0, 5, false, false);
+
+    waitCompleteMs(getTimeoutMillis());
+    currentCommissioner().shutdownSubscription(getNodeId(), currentCommissioner.getFabricId(), subscriptionId);
```

Comment:
Do we need to check `subscriptionId != 0` here before shutting down the subscription? If the subscription fails to establish it won't be set.

Reply:
in c++ layer, we would check whether it is active read client, and decide whether it can be called.
subscriptionId 0 is valid value...